core/src/main/java/org/infinispan/statetransfer/StateTransferManagerImpl.java

```diff
@@ -94,14 +94,16 @@ public void start() throws Exception {
          persistentStateChecksum = Optional.empty();
       }
 
+      float capacityFactor = globalConfiguration.isZeroCapacityNode() ? 0.0f : configuration.clustering().hash().capacityFactor();
```

Comment:
I really wish this was set in the `HashConfigurationBuilder#create` method instead, so we don't have to worry about all the references. But we don't have access to the `GlobalConfiguration` there as far as I know. :(

Reply:
Can I do something here now then ?